src/Entity/BannerSelection/Campaign.php

```diff
@@ -35,7 +35,22 @@ public function getEnd(): \DateTime {
 		return $this->end;
 	}
 
-	public function isActiveAtPointInTime( \DateTime $time ): bool {
-		return $time->getTimestamp() > $this->start->getTimestamp() && $time->getTimestamp() < $this->end->getTimestamp();
+	public function getStart(): \DateTime {
+		return $this->start;
+	}
+
+	public function isInActiveDateRange( \DateTime $time ): bool {
+		return
+			$time->getTimestamp() > $this->getStart()->getTimestamp() &&
+			$time->getTimestamp() < $this->getEnd()->getTimestamp();
+	}
+
+ public function selectBucket( ?string $bucketId, callable $fallbackSelectionStrategy ): Bucket {
```

Comment:
This method is meant to select the bucket from the buckets that are passed in to the campaign. When `$bucketId` is a valid id in `$this->buckets` then return the matching one, otherwise pass the buckets to `$fallbackSelectionStrategy` which will do the choosing.